cmd/certgen.go

```diff
@@ -69,6 +69,16 @@ func New() *cobra.Command {
 	flags.String(option.K8sKubeConfigPath, "", "Path to the K8s kubeconfig file. If absent, the in-cluster config is used.")
 	flags.Duration(option.K8sRequestTimeout, defaults.K8sRequestTimeout, "Timeout for K8s API requests")
 
+	flags.String(option.CACertFile, "", "Path to provided Cilium CA certificate file (required if Cilium CA is not generated)")
+	flags.String(option.CAKeyFile, "", "Path to provided Cilium CA key file (required if Cilium CA is not generated)")
+
+	flags.Bool(option.CAGenerate, defaults.CAGenerate, "Generate and store Cilium CA certificate")
+	flags.Bool(option.CAReuseSecret, defaults.CAReuseSecret, "Reuse the Cilium CA secret if it exists, otherwise generate a new one")
+	flags.String(option.CACommonName, defaults.CACommonName, "Cilium CA common name")
+	flags.Duration(option.CAValidityDuration, defaults.CAValidityDuration, "Cilium CA validity duration")
+	flags.String(option.CASecretName, defaults.CASecretName, "Name of the K8s Secret where the Cilium CA cert and key are stored in")
+	flags.String(option.CASecretNamespace, "", "Overwrites the namespace of the K8s Secret where the Cilium CA cert and key are stored in")
```

Comment:
Yes. I would even be fine if we removed them fully. I don't think `certgen` has any uses outside of the Cilium Helm charts and therefore if the Cilium Helm charts will remove those flags, I don't see a reason to keep them around.

Reply:
sounds good to me, should we also [remove instead of deprecating the hubble ca configmap](https://github.com/cilium/certgen/pull/43) as it has already been removed from Cilium?

Reply:
My understanding based on a previous comment is that the deprecation of the `HubbleCA*` and `ClustermeshApiserverCA*` flags needs to be deprecated first in a Cilium release and then the logic for generating these CAs will be removed in a subsequent release. Based on this conversation it sounds like we are in favor of "ripping off the bandage" and removing the flags and logic all at once. If that's the case I'm fine with removing the flags and logic as part of this PR.

Reply:
Yes, I am very much in favor off just removing the flags and associated logic without prior deprecation. You may do it in this PR, but we can also do it in a separate PR later, whatever seems easier.

Reply:
![Burn it down](https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/hankscorpioflamethrower-1528493255.gif?crop=1xw:1xh;center,top&resize=480:*)

Reply:
I've removed these flags and the generation logic in my latest commit. There might be some redundancy with #43, if we want to merge that PR first I can rebase on top of it.

Reply:
Let's merge this one before as #43 will be easier to rebase.

Reply:
If everything looks good would you mind approving it?